pkg/resource/predicates.go

```diff
@@ -41,20 +41,33 @@ func NewPredicates(fn PredicateFn) predicate.Funcs {
 	}
 }
 
-// TODO(negz): Use separate predicates for resource claims and managed
-// resources once controller-runtime supports this.
-// https://github.com/kubernetes-sigs/controller-runtime/issues/572
-
-// HasClassReferenceKinds accepts objects that reference the supplied
-// non-portable class kind, either directly or indirectly via a portable class
-// kind.
-func HasClassReferenceKinds(c client.Client, oc runtime.ObjectCreater, k ClassKinds) PredicateFn {
+// AnyOf accepts objects that pass any of the supplied the supplied predicate
```

Comment:
This is really cool! Wonder if any upstream k8s projects (likely controller-runtime) would want to add this as part of the `NewControllerManagedBy` chain.

Reply:
I believe they'd need to also be interested in taking on the `NewPredicates` pattern for it to work. I'm on the fence as to whether it's worth contributing; it's a fairly simple pattern.